File: 백준/Silver/10773. 제로/제로.c
#include <stdio.h>

int main(void) {  
   int num;
   scanf("%d", &num);

   int arr[num], a, b = 0, index = 0;
   long sum = 0;

   while(1){
      if(b == num){
         break;
      }
      scanf("%d", &a);

      arr[index] = a;
      if(a == 0){
         index -= 1;
         sum -= arr[index];
      }
      else{
         index += 1;
         sum += a;
      }
      b += 1;
   }
   
   printf("%ld", sum);
   
   return 0;
}

File: 백준/Silver/10773. 제로/제로.py
re_try, sum = 0, 0
a = int(input())
arr = []
while(True):
  if(re_try == a):
      break
  
  num = int(input())
  if(num == 0):
    sum -= arr[len(arr) - 1]
    del arr[len(arr) - 1]
  
  else:
    arr.append(num)
    sum += arr[len(arr) - 1]
  
  re_try += 1

print(sum)